### ISSUE TITLE
deliveryservices_required_capabilities - sort doesn't work

### ISSUE THREAD

**Issue by lbathina** [apache/trafficcontrol#4049]
## I'm submitting a ...

- [X] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

both asc and desc gives results in same order. 

## Expected / new behavior:

sorting should be applied on orderby field.
## Minimal reproduction of the problem with instructions:

https://{{TO_BASE_URL}}/api/{{api_version}}/deliveryservices_required_capabilities?orderby=lastUpdated&sortOrder=desc
https://{{TO_BASE_URL}}/api/{{api_version}}/deliveryservices_required_capabilities?orderby=lastUpdated&sortOrder=asc

**Comment by ocket8888** [comment]
If that's true, it affects more than just this endpoint - nearly all API endpoints use the same subroutine for generating `ORDER BY` clauses.

**Comment by lbathina** [comment]
yes. Asked eric to check on another endpoint

**Comment by mhoppa** [comment]
Ahh looks like our order by only allows fields that are defined in the queryParams map `https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go#L83`

With that `Choose the ordering of the results - must be the name of one of the fields of the objects in the response array` which is used in like all the docs to describe order by is going to be incorrect

**Comment by ZugNZwang** [comment]
Can confirm. On server_server_capabilites endpoint orderBy and sortOrder works on all fields except lastUpdated.

**Comment by ocket8888** [comment]
Yeah. Since that's a `TimeNoMod`, the current implementation is incapable of sorting by it. would be nice, though. There should be an option to pass it arbitrary maps to sort on, though maybe that's one of the many things that got buried deep in the CRUDer.

**Comment by mitchell852** [comment]
ok so `orderby=lastUpdated` doesn't work on across the API? can @lbathina change the title/description of this bug to better reflect that?

**Comment by lbathina** [comment]
done
